control-plane/api-gateway/controllers/gateway_class_config_controller.go

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package controllers
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/hashicorp/consul-k8s/control-plane/api/v1alpha1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	gwv1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
+)
+
+const (
+	gatewayClassConfigFinalizer = "gateway-class-exists-finalizer.consul.hashicorp.com"
+)
+
+type Controller struct {
+	client.Client
+
+	Log logr.Logger
+}
+
+// Reconcile reads the state of an Endpoints object for a Kubernetes Service and reconciles Consul services which
+// correspond to the Kubernetes Service. These events are driven by changes to the Pods backing the Kube service.
```

Comment:
This is a vestige, my friend :P

Reply:
Oh I cannot take credit for that, it is direct copy from old repo.

Reply:
😅

Reply:
I learned a new vocabulary word and also actually updated the comment to be from gateway repo